src/idex.ts

```diff
@@ -43,7 +44,9 @@ export class IdentityResolver {
     this.requestedAttributes = this.idexConfig.requestedAttributes || DEFAULT_REQUESTED_ATTRIBUTES
     this.privacyMode = nonNullConfig.privacyMode ?? false
     this.resolvedIdCookie = nonNullConfig.resolvedIdCookie
-    this.idCookieMode = nonNullConfig.idCookie?.mode ?? 'generated'
+    this.generateIdCookie = nonNullConfig.idCookie?.mode === 'generated'
```

Comment:
Arguably this should be fully decoupled from the idcookie resolution and be a dedicated config setting like `identityResolutionConfig.generateIdCookie`